SpatialGDK/Source/SpatialGDK/Public/Interop/SpatialSender.h

```diff
@@ -82,8 +82,10 @@ class SPATIALGDK_API USpatialSender : public UObject
 	void SendCommandResponse(Worker_RequestId RequestId, Worker_CommandResponse& Response);
 	void SendEmptyCommandResponse(Worker_ComponentId ComponentId, Schema_FieldId CommandIndex, Worker_RequestId RequestId);
 	void SendCommandFailure(Worker_RequestId RequestId, const FString& Message);
-	void SendAddComponent(USpatialActorChannel* Channel, UObject* Subobject, const FClassInfo& Info, uint32& OutBytesWritten);
-	void SendRemoveComponent(Worker_EntityId EntityId, const FClassInfo& Info);
+	void SendAddComponentForSubobject(USpatialActorChannel* Channel, UObject* Subobject, const FClassInfo& Info, uint32& OutBytesWritten);
+	void SendAddComponentForComponentData(Worker_EntityId EntityId, TArray<FWorkerComponentData>& ComponentDatas);
```

Comment:
This is transferring ownership over the underlying data so I think the array should be by value rather than by ref.
Really this is just a plaster over the fact that we don't make ownership over a component object explicit at any point, but it's still an improvement.

Reply:
Also I think in all but one case this is called with something that could just be a temporary constructed in the arguments.

Reply:
Changed to pass by val, and construct things in args